### PR TITLE
Update SB, Next, i18next

### PR DESCRIPTION
**Pull request by sri2k1us**
SSIA.
I followed https://storybook.js.org/blog/storybook-6-migration-guide/ and https://storybook.js.org/docs/react/get-started/setup for storybook migration.

I strongly suggest checking out this PR locally and trying it to make sure I haven't missed anything. Thanks.

### CODE REVIEW

**Review by ash3rz — APPROVED** [review]
LGTM :shipit: 
Code looks good! Went through Storybook and only found the app listing had an issue loading the story (not sure if related to this PR or not).  And went through dev mode, clicked through all the pages, and didn't see any issues.

**Comment by sri2k1us** [comment]
> LGTM :shipit:
> Code looks good! Went through Storybook and only found the app listing had an issue loading the story (not sure if related to this PR or not). And went through dev mode, clicked through all the pages, and didn't see any issues.

Will take a look

**Comment by sri2k1us** [comment]
> > LGTM :shipit:
> > Code looks good! Went through Storybook and only found the app listing had an issue loading the story (not sure if related to this PR or not). And went through dev mode, clicked through all the pages, and didn't see any issues.
> 
> Will take a look

Weird..it works for me 😕

image

File: stories/apps/launch/LoadingError.stories.js [hunk 1]
```diff
@@ -31,4 +27,14 @@ export const LoadingError = () => {
     );
 };
 
-export default { title: "Apps / Launch", decorators: [withKnobs] };
+export default {
+    title: "Apps / Launch",
+    component: AppLaunchStoryBase,
+    argTypes: {
+        customMessage: {
```
Comment (psarando) [inline stories/apps/launch/LoadingError.stories.js:34]:
```suggestion
        "Longer Error Message": {
```

File: stories/apps/launch/LoadingError.stories.js [hunk 2]
```diff
@@ -14,9 +12,7 @@ const errorObject = {
     },
 };
 
-export const LoadingError = () => {
-    const customMessage = boolean("Longer Error Message", false);
-
+export const LoadingError = ({ customMessage }) => {
```
Comment (psarando) [inline stories/apps/launch/LoadingError.stories.js:15]:
```suggestion
export const LoadingError = ({ "Longer Error Message": customMessage }) => {
```

**Review by psarando — CHANGES_REQUESTED** [review]
I like the new way to organize stories and adding `argTypes` to the default export for controls.

But it looks like we have to refine our story organization for this upgrade, in order to get rid of `Duplicate title` warnings and attach controls to the right stories.

File: stories/apps/launch/LoadingError.stories.js [hunk 3]
```diff
@@ -31,4 +27,14 @@ export const LoadingError = () => {
...
+    title: "Apps / Launch",
+    component: AppLaunchStoryBase,
+    argTypes: {
+        customMessage: {
+            control: {
+                type: "boolean",
+            },
+        },
+    },
+};
```
[the remark this answers is not in the document]
Comment (sri2k1us) [reply stories/apps/launch/LoadingError.stories.js:40]: Removing title from the `export` throws an error in SB. I was able re-organize stories in hierarchical fashion by giving unique names to each story. For example, all stories under `/stories/apps/launch` will have titles with this pattern `Apps / Launch / <current story>`. 
image

File: .storybook/preview.js [hunk 4]
```diff
@@ -36,25 +35,26 @@ function MockUserProfile() {
...
                         <CyVerseAnnouncer />
                     </PreferencesProvider>
-                </I18nProviderWrapper>
-            </ReactQueryConfigProvider>
-        </UserProfileProvider>
-    </ThemeProvider>
-));
-
-//redirect console error / logs / warns to action logger
-addDecorator((storyFn, context) => withConsole()(storyFn)(context));
```
[the remark this answers is not in the document]
Comment (sri2k1us) [reply .storybook/preview.js:56]: Good Catch. Added back the `console` add-on.

File: stories/apps/launch/LoadingError.stories.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (psarando) [reply stories/apps/launch/LoadingError.stories.js]: I tried it out in my local copy of this PR, and I also see the error when only removing the `title` from the default exports in these stories.

I think I preferred the old hierarchy, though. This new hierarchy means a lot of extra clicks to navigate and browse from one story to the next.

So in order to preserve the original organization, [we can re-export stories in 1 index file that defines the title](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#deprecated-support-for-duplicate-kinds), then delete all other default exports from the re-exported stories.

Using the stories under `stories/apps/launch` as an example, a `stories/apps/launch/index.stories.js` file can be created like this:
```javascript
export * from "./AgaveWordCount.stories";
export * from "./DeprecatedParamsApp.stories";
export * from "./DEWordCount.stories";
export * from "./FlagParams.stories";
export * from "./InputParams.stories";
export * from "./JupyterLabNoParams.stories";
export * from "./LoadingError.stories";
export * from "./LoadingMask.stories";
export * from "./NumberParams.stories";
export * from "./OutputParams.stories";
export * from "./Pipeline.stories";
export * from "./ReferenceGenomeParams.stories";
export * from "./SelectParams.stories";
export * from "./TextParams.stories";

export default { title: "Apps / Launch" };
```
Then in each of the original story files the `export default { title: "Apps / Launch" };` line will have to be deleted.

For the stories that use controls, the `export default` can be replaced with `<StoryName>.argTypes = {...}` as described here: https://github.com/storybookjs/storybook/blob/next/addons/controls/README.md#knobs-to-manually-configured-args

For example, this `LoadingError` story's default export would be replaced with this:
```javascript
LoadingError.argTypes = {
    customMessage: {
        control: {
            type: "boolean",
        },
    },
};
```

[the remark this answers is not in the document]
Comment (sri2k1us) [reply stories/apps/launch/LoadingError.stories.js]: image
Adding current SB screenshot for comparison. 
It requires the same number of clicks (3 or just 2 in most cases) to access, for example `App / Launch / Loading Error`. Please remember, you don't have to expand the `App / Launch / Loading Error` to load that story, simply clicking on the parent node will load that story.
Anyway, the spirt of the PR is to update the version of the software library without introducing new issues / bugs. I think that is achieved.

[the remark this answers is not in the document]
Comment (psarando) [reply stories/apps/launch/LoadingError.stories.js]: 👌 fair enough.
I also forgot about the auto-selection when expanding a leaf-category.

**Review by psarando — APPROVED** [review]

**Pull request merged**